locallib.php

```diff
@@ -687,29 +687,51 @@ function($u) {
  * Returns an array containing all the roles in a context.
  *
  * @param context $context
+ * @param bool $onlyviewableroles
  *
  * @return array $roles
  */
-function bigbluebuttonbn_get_roles(context $context = null) {
-    $roles = (array) role_get_names($context);
-    foreach ($roles as $key => $value) {
-        $roles[$key] = $value->localname;
+function bigbluebuttonbn_get_roles(context $context = null, bool $onlyviewableroles = true) {
+    global $CFG;
+
+    if ($onlyviewableroles == true && $CFG->branch >= 35) {
```

Comment:
What is that hardcoded branch set to 35?

Reply:
The get_viewable_roles() function was added with MDL-50666 in Moodle 3.5.
That's why the fix of this patch cannot be applied to Moodle core versions before 3.5.
As your plugin releases are covering multiple Moodle core versions, this is the only viable solution.